### PR TITLE
Generic Caching

### PR DESCRIPTION
**Pull request by tbwrigh**

### CODE REVIEW

File: .github/workflows/python-lint.yaml [hunk 1]
```diff
@@ -15,14 +15,9 @@ jobs:
...
-      - run: isort --profile black --check --diff ./
-        working-directory: ./backend
-      - run: mypy ./
-        working-directory: ./backend
-      - run: pylint --rcfile ./backend/pylintrc -E ./backend
-      - run: pytest ./
-        working-directory: ./backend
+      - run: pip install -r ./backend/requirements.txt
+      - run: black --check --diff ./backend/
+      - run: isort --profile black --check --diff ./backend/
```
Comment (OxygenCobalt) [inline .github/workflows/python-lint.yaml:20]: This was intentional. Isort behaves differently depending on if it's aware of pip dependencies and not, so currently isort is not correctly checking 3rd party library import order on the CI.

**Review by OxygenCobalt — CHANGES_REQUESTED** [review]
Overall good

File: backend/src/caching/memcached.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+from pymemcache.client import base
+
+from .cache import Cache
+
+
+class Memcached(Cache):
+    def __init__(self, config={}):
+        self.expire = config["expire"] if "expire" in config else 300
+        if "memcache_address" not in config:
+            raise ValueError("memcache_address not in config")
```
Comment (ezrichards) [inline backend/src/caching/memcached.py:10]: I'm a little wary of 1) these error messages and 2) the fact that we're raising errors anyways; we want to, as they say, crash/fail gracefully. If something goes wrong with the cache, the entire backend could potentially go down.

**Review by ezrichards — APPROVED** [review]
Looks good overall.

**Review by shanecranor — APPROVED** [review]

**Comment by shanecranor** [comment]
looks good assuming you tested it

File: .github/workflows/python-lint.yaml (hunk 1 above)
Comment (OxygenCobalt) [reply .github/workflows/python-lint.yaml:20]: Alright, I'll just leave this as-is and run isort in the broken mode. I'll figure out the issue later in-person.

**Review by OxygenCobalt — APPROVED** [review]

**Review by OxygenCobalt — CHANGES_REQUESTED** [review]
Actually, one change: You need to add the new environment variables to the docker compose file so it doesn't break.

**Review by OxygenCobalt — APPROVED** [review]

**Pull request merged**